### DISCUSSION TITLE
谁能具体解释下random/randomized 和ios/safari的区别啊

### DISCUSSION THREAD

**Discussion by GreatMichaelLee** [XTLS/Xray-core#1628] [General]
RPRX大大的release notes愣是没看懂。。。。

**Comment by FranzKafkaYu** [comment]
很好理解。ios/safari与Chrome, Firefox这些一样，属于确定类型的指纹，如果你fingerprint填写random,那么xray会在ModernFingerprints这个数组内随机选取一个指纹，而如果你fingerprint填写randomized,那么则会完全随机地生成一个指纹，但会保证xtls-rprx-vision与REALITY的使用要求。

**Reply by GreatMichaelLee** [reply]
> 很好理解。ios/safari与Chrome, Firefox这些一样，属于确定类型的指纹，如果你fingerprint填写random,那么xray会在ModernFingerprints这个数组内随机选取一个指纹，而如果你fingerprint填写randomized,那么则会完全随机地生成一个指纹，但会保证xtls-rprx-vision与REALITY的使用要求。

所以随机生成的指纹不属于任何modemfingerprint的类型的指纹是这样理解吧？随机指纹和没有指纹看上去更像伪造的因为不属于常见pattern的指纹特征更容易被GFW识别，是这个逻辑吗？

**Reply by GreatMichaelLee** [reply]
> 很好理解。ios/safari与Chrome, Firefox这些一样，属于确定类型的指纹，如果你fingerprint填写random,那么xray会在ModernFingerprints这个数组内随机选取一个指纹，而如果你fingerprint填写randomized,那么则会完全随机地生成一个指纹，但会保证xtls-rprx-vision与REALITY的使用要求。

ios和safari有什么区别啊？

**Comment by Nyar233** [comment]
random：随机**选取**一个已经设定好的指纹
randomized：随机**生成**一个指纹

**Comment by GreatMichaelLee** [comment]
另外，为啥说ios效果更好，不是理论上应该random更好吗，如果一条流里有多个client指纹，不是看上去更像多个不同的应用clients和server的tls通信，为啥某个单一固定指纹会更安全呢？

**Reply by Felix2yu** [reply]
我的理解是如果用了random以后，ModernFingerprints列表里的那些指纹在大尺度时间范围内会呈现均匀分布，这本身就是一个特征。

**Reply by ttc0419** [reply]
顺便问一下randomized存在意义是什么？当初使用utls不就是要模拟真实浏览器的TLS请求特征么？随机生成指纹不是特征又出来了？

**Reply by Nyar233** [reply]
> 顺便问一下randomized存在意义是什么？当初使用utls不就是要模拟真实浏览器的TLS请求特征么？随机生成指纹不是特征又出来了？

可能是为了对抗针对某些特定指纹的黑名单封锁；可能是为了增加指纹分布的随机性；可能是为了增加用户的可选择面，多一个选择也没什么坏处；可能是因为utls库本身支持随机生成指纹，遂引入。

**Reply by ttc0419** [reply]
> 可能是为了对抗针对某些特定指纹的黑名单封锁；可能是为了增加指纹分布的随机性；

作为代理软件，个人想不出任何理由对主流浏览器指纹封锁。

> 可能是为了增加用户的可选择面，多一个选择也没什么坏处；可能是因为utls库本身支持随机生成指纹，遂引入。

有，这个是一个很危险的选项，会引入特征，一个好的软件应该有foolproof的默认值。

**Reply by Nyar233** [reply]
> > 可能是为了对抗针对某些特定指纹的黑名单封锁；可能是为了增加指纹分布的随机性；
> 
> 作为代理软件，个人想不出任何理由对主流浏览器指纹封锁。
> 

这只是个猜想，源于此 https://github.com/refraction-networking/utls#randomized-fingerprint

> > 可能是为了增加用户的可选择面，多一个选择也没什么坏处；可能是因为utls库本身支持随机生成指纹，遂引入。
> 
> 有，这个是一个很危险的选项，会引入特征，一个好的软件应该有foolproof的默认值。

目前xray-core对指纹的默认值是空，也就是不启用。至于特征的危险性，我觉得选择更多反而更能分散特征；退一步来讲，假如实践证明randomized这个模式是危险的，用户也可以选择不用它，改而使用其他指纹。

**Comment by cccp6** [comment]
Judging from the recent speed limitation incident on reality, some ISPs could be putting QoS on specific fingerprint (chrome). So your assumption is confirmed now. 

> > 顺便问一下randomized存在意义是什么？当初使用utls不就是要模拟真实浏览器的TLS请求特征么？随机生成指纹不是特征又出来了？
> 
> 可能是为了对抗针对某些特定指纹的黑名单封锁；可能是为了增加指纹分布的随机性；可能是为了增加用户的可选择面，多一个选择也没什么坏处；可能是因为utls库本身支持随机生成指纹，遂引入。
